src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -1080,16 +1035,13 @@ private async Task<ChainedHeaderBlock> LoadBlockDataAsync(uint256 blockHash)
                 return chainedHeaderBlock;
             }
 
-            if (this.blockStore != null)
+            Block block = await this.blockStore.GetBlockAsync(blockHash).ConfigureAwait(false);
+            if (block != null)
```

Comment:
Actually no, there is very specific strategy on which blocks are maintained inside of the memory and if we just populated the header with the block, it would be memory leak because there is no code that would remove it from the memory later